source/constructs/api/data_source/service.py

```diff
@@ -2581,29 +2671,356 @@
+    created_jdbc.custom_jdbc_cert = ""
+    created_jdbc.custom_jdbc_cert_string = ""
+    created_jdbc.jdbc_driver_class_name = ""
+    created_jdbc.jdbc_driver_jar_uri = ""
+    created_jdbc.last_updated_time = ""
+    created_jdbc.network_availability_zone = ""
+    # created_jdbc.secret = ""
+    created_jdbc.skip_custom_jdbc_cert_validation = "false"
+    return created_jdbc
+
+async def batch_add_conn_jdbc(created_jdbc_list):
+    tasks = [asyncio.create_task(__add_jdbc_conn_batch(jdbc)) for jdbc in created_jdbc_list]
+    return await asyncio.gather(*tasks)
+
+
+def batch_sync_jdbc(jdbc_list):
+    return asyncio.run(batch_sync_jdbc_manager(jdbc_list))
+
+async def batch_sync_jdbc_manager(jdbc_list):
+    tasks = [asyncio.create_task(__batch_sync_jdbc_worker(jdbc)) for jdbc in jdbc_list]
+    return await asyncio.gather(*tasks)
+
+async def __batch_sync_jdbc_worker(jdbc):
+    sync_jdbc_connection(jdbc)
+
+def __gen_jdbc_targets_from_db_names(connection_name, db_names):
+    jdbc_targets = []
+    for db_name in db_names:
+        trimmed_db_name = db_name.strip()
+        if trimmed_db_name:
+            jdbc_targets.append({
+                'ConnectionName': connection_name,
+                'Path': f"{trimmed_db_name}/%"
+            })
+    return jdbc_targets
+
+def __update_crawler(provider_id, account_id, instance, region, jdbc_targets, crawler_name, glue_database, crawler_role_arn):
+    assume_account, assume_region = __get_admin_info(JDBCInstanceSourceBase(account_provider_id=provider_id,
+                                                                            account_id=account_id,
+                                                                            instance_id=instance,
+                                                                            region=region))
+    try:
+        __get_glue_client(assume_account, assume_region).update_crawler(
+            Name=crawler_name,
+            Role=crawler_role_arn,
+            DatabaseName=glue_database,
+            Targets={
+                'JdbcTargets': jdbc_targets,
+            },
+            SchemaChangePolicy={
+                'UpdateBehavior': 'UPDATE_IN_DATABASE',
+                'DeleteBehavior': 'DELETE_FROM_DATABASE'
+            }
+        )
+    except Exception as e:
+        logger.error(traceback.format_exc())
+        raise BizException(MessageEnum.BIZ_UNKNOWN_ERR.get_code(),
+                           MessageEnum.BIZ_UNKNOWN_ERR.get_msg())
+
+def __get_admin_info(jdbc):
+    account_id = jdbc.account_id if jdbc.account_provider_id == Provider.AWS_CLOUD.value else admin_account_id
+    region = jdbc.region if jdbc.account_provider_id == Provider.AWS_CLOUD.value else admin_region
+    return account_id, region
+
+async def __add_jdbc_conn_batch(jdbc: JDBCInstanceSource):
+    try:
+        add_jdbc_conn(jdbc)
+        return jdbc.account_provider_id, jdbc.account_id, jdbc.region, jdbc.instance_id, "SUCCESSED", None
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are unpacking more than three variables while using a function that returns more than one variable. Using a large number of return values is prone to errors. We recommend that you return a small class or `namedtuple` instance instead.

[Learn more](https://docs.python.org/3/library/typing.html#typing.NamedTuple)

Similar issue at line numbers 2954 and 2957.